scripts/analyze_codebase.py

```
@@ -0,0 +1,284 @@
+    # Generate JSON report if requested
+    if args.json:
+        with open(args.json, "w") as f:
+            json.dump(results, f, indent=2)
+    
+    # Exit with non-zero if issues found
+    total_issues = results["duplicates"]["files"] + results["duplicates"]["blocks"] + results["unreferenced"]["count"]
+    sys.exit(1 if total_issues > 0 else 0)
+
+
+if __name__ == "__main__":
+    main()
```

Ruff (0.12.2)

No newline at end of file

Add trailing newline

(W292)